File: back/back/Models/Aeronef.cs
using LiteDB;

namespace back.Models;

public sealed class Aeronef
{
    [BsonId]
    public int Id { get; set; }

    public int Prix { get; set; }

    public string Nom { get; set; } = null!;
    public string Role { get; set; } = null!;
    public string? Description { get; set; }
}

File: front/src/models/Aeronef.ts
export type Aeronef =
{
    id: number,
    nom: string,
    role: string,
    description: string | null,
    prix: number,
    urlImage: string
}

export type AeronefLeger =
{
    id: number,
    nom: string
}

export type VaisseauAeronefPlaceDisponible = 
{
    id: number,
    nomVaisseau: string,
    nombrePlace: number
}

export type AeronefAchatRequete =
{
    idVaisseauPosseder: number,
    quantite: number
}

export type AeronefRequete = Omit<Aeronef, "id">;